lib/src/progress_bar.dart

```diff
@@ -65,56 +66,75 @@ class _VideoProgressBarState extends State<VideoProgressBar> {
-        if (!controller.value.isInitialized) {
-          return;
-        }
-        _seekToRelativePosition(details.globalPosition);
-
-        widget.onDragUpdate?.call();
-      },
-      onHorizontalDragEnd: (DragEndDetails details) {
-        if (_controllerWasPlaying) {
-          controller.play();
-        }
-
-        widget.onDragEnd?.call();
-      },
-      onTapDown: (TapDownDetails details) {
-        if (!controller.value.isInitialized) {
-          return;
-        }
-        _seekToRelativePosition(details.globalPosition);
-      },
-      child: Center(
-        child: Container(
-          height: MediaQuery.of(context).size.height,
-          width: MediaQuery.of(context).size.width,
-          color: Colors.transparent,
-          child: CustomPaint(
-            painter: _ProgressBarPainter(
-              value: controller.value,
-              colors: widget.colors,
-              barHeight: widget.barHeight,
-              handleHeight: widget.handleHeight,
-              drawShadow: widget.drawShadow,
+    final ChewieController chewieController = ChewieController.of(context);
+
+    return chewieController.draggableProgressBar
+        ? GestureDetector(
+            onHorizontalDragStart: (DragStartDetails details) {
+              if (!controller.value.isInitialized) {
+                return;
+              }
+              _controllerWasPlaying = controller.value.isPlaying;
+              if (_controllerWasPlaying) {
+                controller.pause();
+              }
+
+              widget.onDragStart?.call();
+            },
+            onHorizontalDragUpdate: (DragUpdateDetails details) {
+              if (!controller.value.isInitialized) {
+                return;
+              }
+              _seekToRelativePosition(details.globalPosition);
+
+              widget.onDragUpdate?.call();
+            },
+            onHorizontalDragEnd: (DragEndDetails details) {
+              if (_controllerWasPlaying) {
+                controller.play();
+              }
+
+              widget.onDragEnd?.call();
+            },
+            onTapDown: (TapDownDetails details) {
+              if (!controller.value.isInitialized) {
+                return;
+              }
+              _seekToRelativePosition(details.globalPosition);
+            },
+            child: Center(
+              child: Container(
+                height: MediaQuery.of(context).size.height,
+                width: MediaQuery.of(context).size.width,
+                color: Colors.transparent,
+                child: CustomPaint(
+                  painter: _ProgressBarPainter(
+                    value: controller.value,
+                    colors: widget.colors,
+                    barHeight: widget.barHeight,
+                    handleHeight: widget.handleHeight,
+                    drawShadow: widget.drawShadow,
+                  ),
+                ),
+              ),
```

Comment:
This could be extracted to a Getter and use it between the lines `122-137`
```
Widget get centeredProgressBar => Center(
                child: Container(
                  height: MediaQuery.of(context).size.height,
                  width: MediaQuery.of(context).size.width,
                  color: Colors.transparent,
                  child: CustomPaint(
                    painter: _ProgressBarPainter(
                      value: controller.value,
                      colors: widget.colors,
                      barHeight: widget.barHeight,
                      handleHeight: widget.handleHeight,
                      drawShadow: widget.drawShadow,
                    ),
                  ),
                ),
              );
```